signer/tests/integration/transaction_signer.rs

```diff
@@ -0,0 +1,72 @@
+use signer::storage;
+use signer::testing;
+
+use futures::StreamExt;
+use rand::RngCore;
+use sqlx::Executor;
+
+async fn test_environment(
+    pool: sqlx::PgPool,
+) -> testing::transaction_signer::TestEnvironment<impl FnMut() -> storage::postgres::PgStore> {
+    let num_signers = 3;
+    let context_window = 3;
+    let test_databases: Vec<_> = futures::stream::iter(0..num_signers)
+        .then(|_| async { new_database(&pool).await })
+        .collect()
+        .await;
+
+    let mut idx = 0;
+
+    testing::transaction_signer::TestEnvironment {
+        storage_constructor: move || {
+            idx = (idx + 1) % test_databases.len();
+            storage::postgres::PgStore::from(test_databases.get(idx).unwrap().clone())
+        },
+        context_window,
+        num_signers,
+    }
+}
+
+static MIGRATOR: sqlx::migrate::Migrator = sqlx::migrate!();
+
+async fn new_database(pool: &sqlx::PgPool) -> sqlx::PgPool {
+    let mut rng = rand::rngs::OsRng;
+    let db_name = format!("test_db_{}", rng.next_u64());
```

Comment:
Yeah I thought about having an increasing integer here as well, but I figured that could risk collisions between test runs - because we don't properly clean up the database 😅. This is a bit hacky and not my finest work. I'm also not too happy about using OsRng, but I didn't want to use a seeded rng here for the same reason.

I'd love to work more on this test utility, but I think that might not be the best use of our time right now unfortunately. I think the current hacky solution is decent enough for the time being, and I don't see it getting in our way or cause any problems in the near future.